app/models/manageiq/providers/hawkular/datawarehouse_manager.rb

```diff
@@ -46,10 +46,17 @@ def self.raw_connect(hostname, port, token, alerts = false)
     end
 
     def connect(options = {})
-      @client ||= self.class.raw_connect(hostname,
-                                         port,
-                                         authentication_token('default'),
-                                         options[:alerts])
+      @clients ||= {}
```

Comment:
Right, this is a memoization technique @client is defined only here. On the first call it will create and return the relevant client and from that point on it will return the stored value.